ddtrace/profiling/profiler.py

```diff
@@ -84,7 +84,7 @@ def _restart_on_fork(self):
         # Be sure to stop the parent first, since it might have to e.g. unpatch functions
         # Do not flush data as we don't want to have multiple copies of the parent profile exported.
         try:
-            self._profiler.stop(flush=False)
+            self._profiler.stop(flush=False, join=False)
```

Comment:
why do we not join here?

Reply:
It's an instance where making the tests pass required maintaining the old behavior.

Reply:
Previous behavior did not join (see `_ProfilerInstance._stop_service()`) by default, but in this PR it does now, so we're keeping the same behavior as previously.